src/algebra/group/hom.lean

```diff
@@ -48,75 +48,153 @@ monoid_hom, add_monoid_hom
 variables {M : Type*} {N : Type*} {P : Type*} -- monoids
   {G : Type*} {H : Type*} -- groups
 
-/-- Bundled add_monoid homomorphisms; use this for bundled add_group homomorphisms too. -/
-structure add_monoid_hom (M : Type*) (N : Type*) [add_monoid M] [add_monoid N] :=
+-- for easy multiple inheritance
+set_option old_structure_cmd true
+
+/-- Homomorphism that preserves zero -/
+structure zero_hom (M : Type*) (N : Type*) [has_zero M] [has_zero N] :=
 (to_fun : M → N)
 (map_zero' : to_fun 0 = 0)
+
```

Comment:
It would be neat if we had `→₀` for `zero_hom`, `→₁` for `one_hom`, `→+` for `add_hom`, etc. But again, not very practical.

Reply:
`→₀` is already `finsupp`, isn't it?

Reply:
Exactly, that's half of the problem, and that `→{+, *}` is `{add_}monoid_hom notation is the other half :)

Reply:
We could search/replace `→₀` with `→ᶠ`, which would free up one notation.
We could then use `→ₘ` and `→ₐ` for mul/add maps. But I guess both are already in use by measurable functions / alg-homs. So... probably hopeless.

Reply:
Either way, adding notation sounds like a follow-up for someone eles to explore